File: core/src/main/java/org/toml/ide/formatter/TomlFormattingModelBuilder.java
/*
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package org.toml.ide.formatter;

import consulo.annotation.component.ExtensionImpl;
import consulo.document.util.TextRange;
import consulo.language.Language;
import consulo.language.ast.ASTNode;
import consulo.language.codeStyle.*;
import consulo.language.psi.PsiFile;
import jakarta.annotation.Nonnull;
import jakarta.annotation.Nullable;
import org.toml.lang.TomlLanguage;

@ExtensionImpl
public class TomlFormattingModelBuilder implements FormattingModelBuilder {
    @Nullable
    @Override
    public TextRange getRangeAffectingIndent(PsiFile file, int offset, ASTNode elementAtOffset) {
        return null;
    }

    @Nonnull
    @Override
    public FormattingModel createModel(@Nonnull FormattingContext formattingContext) {
        CodeStyleSettings settings = formattingContext.getCodeStyleSettings();
        var element = formattingContext.getPsiElement();
        TomlFmtContext ctx = TomlFmtContext.create(settings);
        Block block = createBlock(element.getNode(), null, Indent.getNoneIndent(), null, ctx);
        return FormattingModelProvider.createFormattingModelForPsiFile(element.getContainingFile(), block, settings);
    }

    @Nonnull
    @Override
    public Language getLanguage() {
        return TomlLanguage.INSTANCE;
    }

    @Nonnull
    public static ASTBlock createBlock(@Nonnull ASTNode node, @Nullable Alignment alignment, @Nullable Indent indent, @Nullable Wrap wrap, @Nonnull TomlFmtContext ctx) {
        return new TomlFmtBlock(node, alignment, indent, wrap, ctx);
    }
}

File: core/src/main/kotlin/org/toml/ide/formatter/TomlFormattingModelBuilder.kt
/*
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package org.toml.ide.formatter

import consulo.annotation.component.ExtensionImpl
import consulo.document.util.TextRange
import consulo.language.ast.ASTNode
import consulo.language.codeStyle.*
import consulo.language.psi.PsiFile
import org.toml.lang.TomlLanguage

@ExtensionImpl
class TomlFormattingModelBuilder : FormattingModelBuilder {
    override fun getRangeAffectingIndent(file: PsiFile?, offset: Int, elementAtOffset: ASTNode?): TextRange? = null

    override fun createModel(formattingContext: FormattingContext): FormattingModel {
        val settings = formattingContext.codeStyleSettings
        val element = formattingContext.psiElement
        val ctx = TomlFmtContext.create(settings)
        val block = createBlock(element.node, null, Indent.getNoneIndent(), null, ctx)
        return FormattingModelProvider.createFormattingModelForPsiFile(element.containingFile, block, settings)
    }

    override fun getLanguage() = TomlLanguage

    companion object {
        fun createBlock(
            node: ASTNode,
            alignment: Alignment?,
            indent: Indent?,
            wrap: Wrap?,
            ctx: TomlFmtContext
        ): ASTBlock = TomlFmtBlock(node, alignment, indent, wrap, ctx)
    }
}
